core/src/main/java/com/linecorp/armeria/server/ServerConfigHolder.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright 2021 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.util.AbstractListenable;
+
+public final class ServerConfigHolder extends AbstractListenable<ServerConfig> {
```

Comment:
Do we need this holder even though we don't close the previously established connections?

Reply:
I think we will still need to notify all HttpServerHandlers about ServerConfig changes. This AbstractListenable implementation helps with that. If I don't do this I will have to keep a list of all HttpServerHandler inside HttpServerPipelineConfigurator